qemu/tests/blockdev_snapshot_merge.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from provider.blockdev_snapshot_base import BlockDevSnapshotTest
+
+
+class BlockdevSnapshotMergeTest(BlockDevSnapshotTest):
+    def pre_test(self):
+        if not self.main_vm.is_alive():
+            self.main_vm.create()
+        self.main_vm.verify_alive()
+        if self.base_tag != "image1":
+            self.configure_data_disk()
+
+    def prepare_snapshot_file(self):
+        self.params["image_size_%s" % self.snapshot_tag] = self.base_image.size
+        self.params["image_name_%s" % self.snapshot_tag] = 'images/' + self.snapshot_tag
+        snapshot_format = self.params.get("snapshot_format", "qcow2")
+        self.params["image_format_%s" % self.snapshot_tag] = snapshot_format
+        super(BlockdevSnapshotMergeTest, self).prepare_snapshot_file()
+
+    def verify_snapshot(self):
+        if self.main_vm.is_alive():
+            self.main_vm.destroy(free_mac_addresses=False)
+        self.snapshot_image_list = []
+        snapshot = self.snapshot_tags[-1]
+        for item in self.snapshot_tags[-2::-1]:
+            base_image = self.get_image_by_tag(item)
+            snapshot_image = self.get_image_by_tag(snapshot)
```

Comment:
Just fyi.
snapshot image is created by sp_admin, actually, we should call sp_admin.remove_volume(img) to remove it, note that 'img' is a StorageVolume object instead of a QemuImg object, here we didn't get an error because we use the QemuImg object's remove method to remove image, and the path to the snapshot image is the same for both StorageVolume and QemuImg objects.
We might have to keep an eye on this mixed use of different objects in future.

Reply:
I tried sp_admin.remove, but seems that it doesn't take effect(All images not removed). So still in my code, image.remove is used.

Reply:
yes, that's because the sp_admin.remove_volume takes a StorageVolume instead of QemuImg object as its input, you can refer to blockdev_base.target_disk_define_by_params and clean_images.

Reply:
@zhencliu Have updated, please help to review again, thanks.